MonstieWash/Assets/Scripts/MonsterAi/MoodArea.cs

```diff
@@ -0,0 +1,93 @@
+    [Tooltip("Toggle on to make the area reduce effectiveness over frequent use.")] [SerializeField] private bool diminishingEffectiveness; //When repeatedly touched will reduce its effects momentarily.
+    [Tooltip("How fast its effectiveness diminishes if above bool toggled on.")] [SerializeField] private int diminishStrength; //How quickly the effectiveness diminishes.
+    [Tooltip("Produces debug text in console when toggled on.")] [SerializeField] private bool debug;
+
+    //Internal states
+    private float currentCooldown;
+    private float currentEffectiveness; //Current effectiveness of area.
+
+    [Serializable]
+    public struct moodEffect
+    {
+        public MoodType mt; //Which mood to target.
+        public float reactionStrength; //How much the mood will change by.
+    }
+
+    private void OnEnable()
+    {
+        //Assign to Input system.
+        InputManager.Inputs.OnActivate_Held += TestTouch;
+
+        m_mb = FindFirstObjectByType<MonsterBrain>();
+        m_ph = FindFirstObjectByType<PlayerHand>();
+        currentEffectiveness = 100;
+    }
+
+
+    private void Update()
+    {
+        //Restore Effectiveness over time.
+        if (diminishingEffectiveness) currentEffectiveness = Mathf.Clamp(currentEffectiveness +( (diminishStrength/2) * Time.deltaTime), 0 , 100);
+        //Reduce Cooldown over time.
+        currentCooldown = Mathf.Clamp(currentCooldown -= Time.deltaTime, 0, areaCooldown);
+    }
+
+
+    /// <summary>
+    /// Function that is called when a touch is detected by the area.
+    /// </summary>
+    void OnTouch()
+    {
+            currentCooldown = areaCooldown; //Reset Cooldown
+
+            //Affect Moods
+            foreach (moodEffect me in moodEffects)
+            {
+                m_mb.UpdateMood(me.reactionStrength * (currentEffectiveness/100), me.mt);
+                if (debug) print($"Reaction Strength  {me.reactionStrength}  at effectivness of {currentEffectiveness} for the mood {me.mt.MoodName}");
+            }
+
+            //Apply diminishing effect if toggled on.
+            if (diminishingEffectiveness) currentEffectiveness = Mathf.Clamp(currentEffectiveness -= diminishStrength, 0, 100);
+
+    }
+
+    /// <summary>
+    /// Called by on activate held from input system, checks if the player hand is overlapping the moodArea.
+    /// </summary>
+    private void TestTouch()
+    {
+        //Skip test if area still on cooldown;
+        if (currentCooldown > 0) return;
+
+        //Check if playerhand is over a mood area and it matches this mood area.
+        var colCheck = Physics2D.OverlapCircle(m_ph.transform.position, 0.1f, layerMask, -999999, 999999);
+        if (colCheck != null)
+        {
+            if (colCheck.gameObject != this.gameObject) return;
+        }
+        else return;
+        
+        //Call On Touch effect
+        OnTouch();
+    }
```

Comment:
For this one there's a few things. The main one is (regarding the comment underneath) that to get the position of the tip of a tool it's fairly straightforward, but you will need to add a getter in the ToolSwitcher.cs just above the getter that's already there like the following:
```
    public List<GameObject> ToolInstances { get { return m_toolInstances; } }
```

Once you've done that, its fairly easy to get the tip. I'll show you the code for it below along with a simplification I made for the final checks.
```suggestion
    private void TestTouch()
    {
        //Skip test if area still on cooldown;
        if (currentCooldown > 0f) return;

        var toolSwitcher = m_ph.GetComponent<ToolSwitcher>();
        Vector3 toolTipPos = m_ph.transform.position;
        //Get the tool tip as long as you don't have an empty hand
        if (toolSwitcher.CurrentToolIndex != -1) toolTipPos = 
        toolSwitcher.ToolInstances[toolSwitcher.CurrentToolIndex].transform.GetChild(0).position;

        //Check if playerhand is over a mood area and it matches this mood area.
        var colCheck = Physics2D.OverlapCircle(toolTipPos, 0.1f, layerMask, -999999, 999999);

        if (colCheck != null && colCheck.gameObject == gameObject)
        {
            //Call On Touch effect
            OnTouch();
        }
    }
```